tasks/windows.ps1

```diff
@@ -0,0 +1,56 @@
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory = $True)]
+  [String]
+  $Master,
+
+  [Parameter(Mandatory = $False)]
+  [String]
+  $CACert_Content,
+
+  [Parameter(Mandatory = $False)]
+  [ValidateScript({ if ([System.URI]::CheckHostName($_) -eq 'Unknown') {
+      throw "Invalid DNS name `"$_`""
+  }; $True})]
+  [String]
+  $CertName,
+
+  [Parameter(Mandatory = $False)]
+  [ValidateScript({ $_ -split ',' | % {
+    if ([System.URI]::CheckHostName($_) -eq 'Unknown') {
+      throw "Invalid DNS name `"$_`""
+  }}; $True})]
+  [String]
+  $DNS_Alt_Names
+)
+
+function Get-HostName
+{
+  $ipAddress = ([System.Net.Dns]::GetHostEntry([System.Environment]::MachineName)).AddressList |
+    ? { $_.AddressFamily -eq 'InterNetwork' } |
+    Select -ExpandProperty IPAddressToString
+
+  [System.Net.Dns]::GetHostEntry($ipAddress).HostName
+}
+
+if (!$PSBoundParameters.ContainsKey('CertName'))
+{
+  $CertName = Get-HostName
+}
+
+if ($CACert_Content) {
+  New-Item -ItemType Directory -Force -Path "$env:ProgramData\PuppetLabs\puppet\etc\ssl\certs"
+  Out-File -InputObject $CACert_Content -FilePath "$env:ProgramData\PuppetLabs\puppet\etc\ssl\certs\ca.pem" -Encoding ascii -Force
+}
+
+[Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
+$webClient = New-Object System.Net.WebClient
+$webClient.DownloadFile("https://${Master}:8140/packages/current/install.ps1", $env:temp + '\install.ps1')
+
+$installerArgs = @("agent:certname='$CertName'")
+if ($PSBoundParameters.ContainsKey('DNS_Alt_Names')) {
+  $installerArgs += "agent:dns_alt_names='$DNS_Alt_Names'"
+}
+
+&($env:temp + '\install.ps1') $installerArgs
```

Comment:
I'm getting an error running this, but having a little trouble tracking it down. The install log says installation was successful, but I get an error that results in the task exiting 1
```
 : System.Management.Automation.RemoteException: [1;33mWarning: No section specified; defaulting to 'main'.
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException
```